ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -295,7 +293,8 @@ impl<'a> TraceBuilder<'a> {
         };
         let expect = *succ_bb == nextbb.block_idx();
         let guard = jit_ir::GuardInstruction::new(jit_ir::Operand::Local(cond), expect);
-        Ok(self.jit_mod.push(guard.into()))
+        let _ = self.jit_mod.push(guard.into());
```

Comment:
This is slightly odd on instructions that don't define a variable like here. Though I don't currently have a suggestion on how to make this better.

Reply:
Presumably a `push_no_rtn` or similar function that panics if it doesn't define a variable?

Reply:
Indeed. I don't have a suggestion either. But there are some upsides that may pay off this oddity?

Reply:
FWIW, there are no upsides to this, and it looks to me like it swallows errors, so there are some downsides? Maybe I've misunderstood.

Reply:
No upsides! OK, my commit message failed to explain then.

Consider this hunk:
```diff
@@ -651,11 +643,15 @@ mod tests {
         let i64_ty_idx = jit_mod
             .type_idx(&jit_ir::Type::Integer(IntegerType::new(64)))
             .unwrap();
-        jit_mod.push(jit_ir::LoadTraceInputInstruction::new(0, i64_ty_idx).into());
-        jit_mod.push(jit_ir::LoadTraceInputInstruction::new(64, i64_ty_idx).into());
-        let op1 = jit_ir::Operand::Local(jit_ir::InstrIdx::new(0).unwrap());
-        let op2 = jit_ir::Operand::Local(jit_ir::InstrIdx::new(1).unwrap());
-        jit_mod.push(jit_ir::AddInstruction::new(op1, op2).into());
+        let ti_op1 = jit_mod
+            .push(jit_ir::LoadTraceInputInstruction::new(0, i64_ty_idx).into())
+            .unwrap()
+            .to_operand(&jit_mod);
+        let ti_op2 = jit_mod
+            .push(jit_ir::LoadTraceInputInstruction::new(64, i64_ty_idx).into())
+            .unwrap()
+            .to_operand(&jit_mod);
+        let _ = jit_mod.push(jit_ir::AddInstruction::new(ti_op1, ti_op2).into());
         let patt_lines = [
             "...",
             "; Add %0, %1",
```

Before we push the instructions, then manually magic up the operands with the correct instruction indices that correspond with the pushed instructions. That is a) laborious, b) error prone.

After, the pushes themselves give us everything we need to make the operands automatically.

Reply:
It kind of does. It errors if the instruction index that would result wasn't expressible in the bits we have.

That's only really a problem if a) the instruction defined a value (it may not), b) the value gets used (in which case the error would either be consumed, or the user would magic up another `InstrIdx` with the same value and consume the same error that would have otherwise been generated).

I'm not sure there's a better way, but I'm open to suggestions.

Reply:
I'm getting cold feet about this change now.

Maybe we should back that commit out, and you just have to be careful with what `InstrIdx`s you magic up? It only really affects unit tests anyway. The trace builder doesn't work in a way that can take advantage of this idiom anyway.

Reply:
Another idiom we could use in tests is something like:
```
self.push(some_instruction);
let instr_idx = self.previous_instr_idx();
```

of course this is more error prone too...

Reply:
If we only need the index in tests, add a second function with `#[cfg(test)]` or similar perhaps?

Reply:
There's currently one non-test call site where the function could be useful.

I think it'd be confusing to have a different API available for testing so, if it's OK with you I will just revert this change.

Reply:
OK.

Reply:
Reverted in 6409770a974abce2a137b21c190a99069f73c8ce